src/jira-connector.ts

```diff
@@ -22,31 +23,44 @@ export class JiraConnector {
     });
   }
 
-  async getTicketDetails(key: string): Promise<JIRADetails> {
-    const issue: JIRA.Issue = await this.getIssue(key);
-    const {
-      fields: { issuetype: type, project, summary },
-    } = issue;
+  async getTicketDetails(key: string, pr_status: string): Promise<void> {
+    try {
+      const url = `/issue/${key}?fields=project,summary,issuetype`;
+      const response = await this.client.get<JIRA.Issue>(url);
+      const issue: JIRA.Issue = response.data;
+      const {
+        fields: { issuetype: type, project, summary },
+      } = issue;
 
-    return {
-      key,
-      summary,
-      url: `${this.JIRA_BASE_URL}/browse/${key}`,
-      type: {
-        name: type.name,
-        icon: type.iconUrl,
-      },
-      project: {
-        name: project.name,
-        url: `${this.JIRA_BASE_URL}/browse/${project.key}`,
-        key: project.key,
-      },
-    };
-  }
-
-  async getIssue(id: string): Promise<JIRA.Issue> {
-    const url = `/issue/${id}?fields=project,summary,issuetype`;
-    const response = await this.client.get<JIRA.Issue>(url);
-    return response.data;
+      const details = {
+        key,
+        summary,
+        url: `${this.JIRA_BASE_URL}/browse/${key}`,
+        type: {
+          name: type.name,
+          icon: type.iconUrl,
+        },
+        project: {
+          name: project.name,
+          url: `${this.JIRA_BASE_URL}/browse/${project.key}`,
+          key: project.key,
+        },
+      };
+      const githubConnector = new GithubConnector();
+      await githubConnector.updatePrDetails(details);
+    } catch (error) {
+      if (error.response) {
+        console.log('Something went wrong in fetching the Jira issues!');
+        console.log('Error code - ' + error.response.status);
+        console.log(error.response.data);
+      }
+      if (pr_status === 'true') {
+        console.log('Setting PR status as failed');
+        process.exit(1);
+      } else {
+        console.log('Ignore the errors and set PR status as success');
+        process.exit(0);
+      }
```

Comment:
lets not mix logic of fetching issue value from jira and failing action and move all error handling which requires the new flag into one place, here
https://github.com/cakeinpanic/jira-description-action/pull/12/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR42

so we would not need to pass extra params to methods and there would be centralised exit point

Reply:
All the action failing is moved to `main.ts` in the catch block. Please take a look.